### ISSUE TITLE
[BUG]X22i no conected to QZ

### ISSUE THREAD

**Issue by bisoft75** [cagnulein/qdomyos-zwift#2937]
QZ no conected, so zwift ask me for change manu
[debug-Wed_Dec_25_12_21_02_2024.log](https://github.com/user-attachments/files/18245706/debug-Wed_Dec_25_12_21_02_2024.log)

**Smartphone (please complete the following information):**
 - Device: Poco X4 GT
 - OS: Android
 - Version 14

**Comment by cagnulein** [comment]
It seems that the treadmills tablet is not logging the metrics 

Can you try to install this instead on the tablet https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12392144945/artifacts/2337179123

**Comment by bisoft75** [comment]
I couldn't install apk (usb debugging is switched and unknown origins into
security also). But I have tried to restart the treadmill with hard reset
with the same result.
QZ into android not receive data (only in 1 time....I don't know what) from
the treadmill. I think troubles came in when I logged in with zwift user
into the QZ app. I have deleted user and pass and run only for a few
seconds....But I don't really sure of the reason.

regards

**Comment by cagnulein** [comment]
You have to unzip that file first 

If you want you can directly install this https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.6.0/QZCompanionNordictrackTreadmill.apk

And enable the ocr flag in the companion 

Let me know!

**Comment by bisoft75** [comment]
I cannot this apk into treadmill, even when I download the file, ifit
automatically update and downloads folder come back empty eacht time.

So first installation QZ app in android read stats from the treadmill, but
in any moment it stopped working, so I think when I log in zwift.


Is it possible reinstall app in my android mobile device?

**Comment by cagnulein** [comment]
You can reinstall it on the phone but it's not the issue.

Do a factory reset on the treadmill and redo the installation process getting the new installation package, so you will install the new apk 3.6.0 of the companion

**Comment by bisoft75** [comment]
I have been able to install apk (ocr flag) but qz companion don't tell me
stats (I have reinstalled also).
regards

**Comment by cagnulein** [comment]
enable both ocr flag and debug flag there and send me a debug log form qz then so i can check which is the issue.

**Comment by cagnulein** [comment]
the log check box is in the companion app in the treadmill
the log generated is instead on the phone as you did on the first post here

**Comment by bisoft75** [comment]
Add logs
[debug-Sun_Dec_29_16_27_06_2024.log](https://github.com/user-attachments/files/18270081/debug-Sun_Dec_29_16_27_06_2024.log)

**Comment by cagnulein** [comment]
@bisoft75 It's a corrupted one : did you close qz before sending it?

**Comment by bisoft75** [comment]
Sorry, try again
[debug-Sun_Dec_29_16_46_01_2024.log](https://github.com/user-attachments/files/18270122/debug-Sun_Dec_29_16_46_01_2024.log)
[debug-Sun_Dec_29_16_47_04_2024.log](https://github.com/user-attachments/files/18270123/debug-Sun_Dec_29_16_47_04_2024.log)

**Comment by cagnulein** [comment]
Mmm i don't see the ocr lines, that's the issue. Is the new version asking for recording the screen? Did you allow it? It's mandatory for this

**Comment by bisoft75** [comment]
Yeah, but every time the output is error.  incan restart treadmill
![17354880232473855873576092763800](https://github.com/user-attachments/assets/25897170-4aee-4173-863f-2c7fd3453cef)
![Uploading 17354880523448347657223111500890.jpg…]()

**Comment by bisoft75** [comment]
Try this. Some way treadmill android says qz Sync is active, so I have some dubts.
![1000056481](https://github.com/user-attachments/assets/39aeb1cd-37e3-4fad-838c-f7bcbba35d6e)
[debug-Sun_Dec_29_17_11_07_2024.log](https://github.com/user-attachments/files/18270219/debug-Sun_Dec_29_17_11_07_2024.log)

**Comment by cagnulein** [comment]
Ok so is the same problem of the #2878

Are you able to gain access to adb to check the adb logcat?

**Comment by cagnulein** [comment]
You can read the notes here https://github.com/cagnulein/qdomyos-zwift/issues/2878#issuecomment-2563809194

**Comment by bisoft75** [comment]
as happen in with [#2878](https://github.com/cagnulein/qdomyos-zwift/issues/2878) S22i  I have tried adb debugging...and another ways as you ask to Maverik (I have attached results).
In my treadmill update is not V2, so I tried ifit2 tools but at least I cannot connect QZ betwin APP&Treadmill
<img width="374" alt="adb-debug" src="https://github.com/user-attachments/assets/fafc1a96-2b98-4845-890d-eab6a55a93f6" />
<img width="432" alt="QZ-Companion-simple" src="https://github.com/user-attachments/assets/3933abc1-5e9b-42d3-8d9e-76462e6bf806" />
<img width="370" alt="script" src="https://github.com/user-attachments/assets/5d01d848-9a74-48e8-bf8a-24a3785384b2" />
[debug-Sat_Jan_4_15_21_57_2025.log](https://github.com/user-attachments/files/18307220/debug-Sat_Jan_4_15_21_57_2025.log)
[debug-Sat_Jan_4_15_16_29_2025.log](https://github.com/user-attachments/files/18307223/debug-Sat_Jan_4_15_16_29_2025.log)

**Comment by cagnulein** [comment]
@bisoft75 are you able to do a factory reset of the treadmill inhibt ifit to update itself? so we should be able to make it work (ifit is trying to close every doors )
